DMCompiler/DM/Visitors/DMObjectBuilder.cs

```diff
@@ -71,10 +71,32 @@ class DMObjectBuilder {
             DMVariable variable;
 
             _currentObject = DMObjectTree.GetDMObject(varDefinition.ObjectPath);
+            //DMObjects store two bundles of variables; the statics in GlobalVariables and the non-statics in Variables.
+            //Lets check if we're duplicating a definition, first.
+            if (_currentObject.HasGlobalVariable(varDefinition.Name))
+            {
+                DMCompiler.Error(new CompilerError(varDefinition.Location, $"Duplicate definition of static var \"{varDefinition.Name}\""));
+                variable = _currentObject.GetGlobalVariable(varDefinition.Name);
+            }
+            else if (_currentObject.HasLocalVariable(varDefinition.Name)) // Overriding a var with a static var
```

Comment:
This comment isn't necessarily correct, the var might not be static.

This will be good to merge after this is changed,

Reply:
third time's the charm